client/components/site-selector/index.jsx

```diff
@@ -185,7 +185,7 @@ const SiteSelector = React.createClass( {
 					href={ this.props.siteBasePath ? siteHref : null }
 					key={ 'site-' + site.ID }
 					indicator={ this.props.indicator }
-					onSelect={ this.onSiteSelect.bind( this, site.slug ) }
+					onSelect={ this.onSiteSelect.bind( this, site.slug, site.ID ) }
```

Comment:
When changing this line, could you also refactor binding of `onSiteSelect` so it happens once in constructor and not in JSX?